src/mongo_client.ts

```diff
@@ -500,6 +501,16 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
 
       Promise.all(activeSessionEnds)
         .then(() => {
+          // If we would attempt to select a server and get nothing back we short circuit
+          // to avoid the server selection timeout.
+          const selector = readPreferenceServerSelector(ReadPreference.primaryPreferred);
+          const topologyDescription = this.topology.description;
+          const serverDescriptions = Array.from(topologyDescription.servers.values());
+          const servers = selector(topologyDescription, serverDescriptions);
+          if (servers.length === 0) {
+            return callback();
+          }
```

Comment:
I think we want to return here, could be wrong, but the callback should only be referenced in the last "then" of the chain. 

So actually a few lines down where we do: 
```ts
if (this.topology == null) {
  return callback();
}
```
Is a teeny bug

Reply:
I've updated to assert there is only 1 session in the pool and changed the check to just return.